sys/include/net/ieee802154/radio.h

```diff
@@ -0,0 +1,1085 @@
+     * function can be polled.
+     *
+     * @pre the device is on
+     * @pre call to @ref ieee802154_radio_ops::request_transmit was successful.
+     *
+     * @post @ref ieee802154_radio_ops::request_set_trx_state MUST be called
+             in order to set the new transceiver status.
+     *
+     * @param[in] dev IEEE802.15.4 device descriptor
+     * @param[out] info the TX information. Pass NULL
+     * if the information is not needed. If the radio supports AutoCCA, the
+     * status should indicate transmission done or channel busy. If the radio
+     * supports frame retransmissions, the status should indicate if medium
+     * was busy, no ACK was received or transmission succeeded.
+     *
+     * @return 0 on success
+     * @return -EAGAIN if the transmission has not finished yet.
+     * @return negative errno on error
+     */
+    int (*confirm_transmit)(ieee802154_dev_t *dev, ieee802154_tx_info_t *info);
+
+    /**
+     * @brief Get the length of the received PSDU frame.
+     *
+     * @pre the device is on
+     * @pre the radio already received a frame (e.g
+     *      @ref ieee802154_dev_t::cb with @ref IEEE802154_RADIO_RX_DONE).
+     *
+     * @post the frame buffer is still protected against new frame arrivals.
+     *
+     * @param[in] dev IEEE802.15.4 device descriptor
+     *
+     * @return length of the frame
+     */
+    int (*len)(ieee802154_dev_t *dev);
+
+    /**
+     * @brief Process the RX done indication
+     *
+     * This function reads the received frame from the internal framebuffer.
+     * It should try to copy the received frame into @p buf and
+     * then unlock the framebuffer (in order to be able to receive more
+     * frames).
+     *
+     * @pre the device is on
+     *
+     * @post @ref ieee802154_radio_ops::request_set_trx_state MUST be called
+             in order to set the new transceiver status.
+     *
+     * @param[in] dev IEEE802.15.4 device descriptor
+     * @param[out] buf buffer to write the received PSDU frame into. If NULL,
+     *             the frame is not copied.
+     * @param[in] size size of @p buf
+     * @param[in] info information of the received frame (LQI, RSSI). Can be
+     *            NULL if this information is not needed.
+     *
+     * @return number of bytes written in @p buffer (0 if @p buf == NULL)
+     * @return -EINVAL if the CRC is not valid. Frame is dropped immediately.
+     * @return -ENOBUFS if the frame doesn't fit in @p
+     */
+    int (*indication_rx)(ieee802154_dev_t *dev, void *buf, size_t size,
+                         ieee802154_rx_info_t *info);
+
+    /**
+     * @brief Turn off the device
+     *
+     * @param[in] dev IEEE802.15.4 device descriptor
+     *
+     * @post the transceiver state is @ref IEEE802154_TRX_STATE_TRX_OFF
+     *
+     * @return 0 on success
+     * @return negative errno on error
+     */
+    int (*off)(ieee802154_dev_t *dev);
```

Comment:
What's the reason behind the asymmetry between `off` and `request_on` / `confirm_on`?
Is `off` supposed to block until the device is turned off?

Reply:
Most radios do this automatically. But I know some radios might take more time. Should we add a `confirm_off` counterpart?

Reply:
We can also just specify that when the function returns, the radio shall be off.

Reply:
https://github.com/RIOT-OS/RIOT/pull/14371/commits/bdbe967f0ceea36fa103518d5e6ba804e7478497